src/auth/userSession.ts

```diff
@@ -229,19 +229,22 @@ export class UserSession {
    * @returns {Promise} that resolves to the user data object if successful and rejects
    * if handling the sign in request fails or there was no pending sign in request.
    */
-  handlePendingSignIn(authResponseToken: string = this.getAuthResponseToken()) {
-    const transitKey = this.store.getSessionData().transitKey
-    const nameLookupURL = this.store.getSessionData().coreNode
+  async handlePendingSignIn(
+    authResponseToken: string = this.getAuthResponseToken()
+  ): Promise<UserData> {
+    const sessionData = await this.store.getSessionData()
+    const transitKey = sessionData.transitKey
+    const nameLookupURL = sessionData.coreNode
     return authApp.handlePendingSignIn(nameLookupURL, authResponseToken, transitKey, this)
   }
 
   /**
    * Retrieves the user data object. The user's profile is stored in the key [[Profile]].
-   * 
-   * @returns {Object} User data object.
+   *
+   * @returns {Promise} User data object.
    */
-  loadUserData() {
-    const userData = this.store.getSessionData().userData
+  async loadUserData(): Promise<UserData> {
```

Comment:
Same concern here as above comment about API breakage

Reply:
Same with this function as the above comment -- default to sync, but take async path when optional param is provided.